test/e2e_node/podresources_test.go

```diff
@@ -0,0 +1,380 @@
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName: "pod-01",
+		},
+	})
+	expectPodResources(cli, 0, 2+1) // test pods + sriovdp
+	tpd.deletePodsForTest(f)
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when only a subset of pods require resources")
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName:        "pod-01",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+		{
+			podName: "pod-02",
+		},
+		{
+			podName:        "pod-03",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+	})
+	expectPodResources(cli, 2, 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+	tpd.deletePodsForTest(f)
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when creating pods which require resources between calls")
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName:        "pod-01",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+		{
+			podName: "pod-02",
+		},
+	})
+	podResources, noResources = getPodResources(cli)
+	framework.ExpectEqual(len(podResources), 1)
+	framework.ExpectEqual(len(noResources), 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName:        "pod-03",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+	})
+	podResources, noResources = getPodResources(cli)
+	framework.ExpectEqual(len(podResources), 2)
+	framework.ExpectEqual(len(noResources), 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+	tpd.deletePodsForTest(f)
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when deleting pods which require resources between calls")
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName:        "pod-01",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+		{
+			podName: "pod-02",
+		},
+		{
+			podName:        "pod-03",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+	})
+	podResources, noResources = getPodResources(cli)
+	framework.ExpectEqual(len(podResources), 2)
+	framework.ExpectEqual(len(noResources), 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+
+	tpd.deletePod(f, "pod-01")
+	podResources, noResources = getPodResources(cli)
+	framework.ExpectEqual(len(podResources), 1)
+	framework.ExpectEqual(len(noResources), 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+	tpd.deletePodsForTest(f)
+}
+
+func podresourcesGetAllocatableResourcesTests(f *framework.Framework, cli kubeletpodresourcesv1.PodResourcesListerClient, sd *sriovData) {
+	ginkgo.By("checking the devices known to the kubelet")
+	resp, err := cli.GetAllocatableResources(context.TODO(), &kubeletpodresourcesv1.AllocatableResourcesRequest{})
+	framework.ExpectNoError(err)
+	devs := resp.GetDevices()
+	cpuIDs := resp.GetCpuIds()
+
+	onlineCPUIds, err := getOnlineCPUs()
```

Comment:
i see this covers the all assignable cpus case, but nothing appears to check the actual cpu assignment for a pod is popuated.  is that intended to come later?

Reply:
yes, but since I need to work to address the other comments I'll try to squeeze it in.